### PR TITLE
Recipe app group project

### PR DESCRIPTION
**Pull request by willy836**
**Milestones**
1. You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
2. Login page and registration page:
    - Should be built with Devise.
3. Food list (only for a group that has less than 3 members):
    - Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
4. Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
5. Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
6. Recipe details:
    - Should display a toggle button that allows for a recipe to be made public or private.
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
7. General shopping list view (only for a group that has less than 3 members):
    - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with 
       the general food list of that user).
    - Should count the total food items and total price of the missing food.
8. Make sure there are no N+1 queries happening.
9. Create a navigation menu that allows users to open all of the pages you created.
10. Write unit and integration tests

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  </div>
+  <div class="recipe-toggler">
+    <% if @recipe.public %>
+      <p>Public</p>
+      <%= button_to "Change public to private", recipe_path(@recipe), method: :patch%>
+    <% else %>
+      <p>Private</p>
+      <%= button_to "Change public to public", recipe_path(@recipe), method: :patch%>
+    <% end %>
+  </div>
```
Comment (addod19) [inline app/views/recipes/show.html.erb:30]: - [ ] In order to match the given wireframe, let us kindly position the toggle button on the right side of the page as we can see below compared to ours
Screenshot from 2023-04-21 15-22-17
Screenshot from 2023-04-21 15-22-39

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  <% if @recipe.user_id == current_user.id %>
+    <section class="bottom-links">
+      <%= link_to 'Generate Shopping List', general_shopping_lists_path, class: "btn-add" %>
+      <%= link_to 'Add ingredient', new_recipe_recipe_food_path(@recipe.id), class: "btn-add" %>
```
Comment (addod19) [inline app/views/recipes/show.html.erb:35]: - [ ] Also the `Add Ingredient` button should lead to the form that allows the user to add a new food, however, currently the app only allow us to add a new quantity but not a new food as we can see below
Screenshot from 2023-04-21 15-25-02
[Screencast from 2023-04-21 03:41:40 EW.webm](https://user-images.githubusercontent.com/46388113/233677289-b86c36da-4289-4a1d-9984-69dbe1edf6fc.webm)

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @willy836 and @Ellon-M, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] README file is professional and has all the relevant headers :fireworks: 
- [ ] All unit and integration test cases passed :100: 
- [ ] Associations between models done correctly :rocket: 
- [ ] Strong parameters used :+1: 
- [ ] Only necessary routes are used :1st_place_medal: 
- [ ] Correct columns and data types used and foreign keys have indexes :fireworks: 
- [ ] Login and registration page uses Devise :rocket: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Going forward, let us kindly capitalize all our commit messages, it makes our repo looks professional and highlights attention to detail

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb (hunk 2 above)
Comment (Ellon-M) [reply app/views/recipes/show.html.erb:35]: The `Add ingredient` button is unrelated to the functionality of adding new food. The `Add Food` button at the top is what performs that action to lead the user to the form to add new food. :+1:

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (Ellon-M) [reply app/views/recipes/show.html.erb:30]: Change noted and made.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @willy836 and @Ellon-M, 

To highlight 👍🏽 
✔️ Design matches wireframe
✔️ No errors

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**